format/spec.md

```diff
@@ -1314,6 +1324,25 @@ Default values are added to struct fields in v3.
 
 Types `timestamp_ns` and `timestamptz_ns` are added in v3.
 
+Sort order and partition field with multiple source field ids are enabled in V3.
+
+Writing V3 metadata:
+
+* Partition field JSON:
+    * `source-ids` was added and is required
+    * `source-id` is no longer required and should be omitted; always use `source-ids` instead
+* Sort order JSON:
+    * `source-ids` was added and is required
+    * `source-id` is no longer required and should be omitted; always use `source-ids` instead
+
+Reading older version metadata for V3:
+
+* Partition field and sort order field `source-ids` must default to a single-value list of the value of `source-id`
+
+Writing older version metadata:
+
+* Partition field and sort order field `source-ids` should be omitted; use `source-id` instead (multiple values not allowed)
```

Comment:
This would prevent v1 and v2 from using multi-arg transforms, but I think that should be supported and should be chosen by the client. See my suggestion above, which explains what to do for compatibility.

Reply:
Yes, initially was thinking to have a hidden flag to allow V1 and V2 to behave like v3, but you are right we should call it out explicitly.  Updated.